pkg/resources/kafka/kafka.go

```diff
@@ -1223,3 +1227,53 @@ func generateServicePortForEListeners(listeners []v1beta1.ExternalListenerConfig
 	}
 	return usedPorts
 }
+
+func (r *Reconciler) reconcileHeadlessService(log logr.Logger) error {
```

Comment:
I would divide this part into two function like: removeServices, removeHeadlessService and use it here: https://github.com/banzaicloud/koperator/blob/815d07e9ecc68defb6c966ba1c838b08fec6d994/pkg/resources/kafka/kafka.go#L153 and here: https://github.com/banzaicloud/koperator/blob/815d07e9ecc68defb6c966ba1c838b08fec6d994/pkg/resources/kafka/kafka.go#L159
What do you think?

Reply:
Sorry for the late.

But in the reconsiling the old kafka services are still in use, so we should not delete it until the end of reconciling.